src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2709,6 +2757,9 @@ internal void DoCommand(EventCommandEventArgs commandArgs)
         /// </summary>
         internal bool EnableEventForDispatcher(EventDispatcher? dispatcher, EventProviderType eventProviderType, int eventId, bool value)
         {
+            if (!IsSupported)
```

Comment:
I wonder if we should do the same if check when methods are void and just do `if (!isSupported) return;`